src/main/java/com/google/step/snippet/external/W3SchoolsClient.java

```diff
@@ -60,9 +63,18 @@ public Card search(String w3Link, String query) {
     String code =
         Jsoup.clean(
             snippets.first().getElementsByClass(CODE_CLASS).text(), Whitelist.basicWithImages());
```

Comment:
A concerning issue here, that I don't have all the answers to cause I don't really understand Jsoup.clean that well, but here goes... For the ["css style"](https://www.w3schools.com/html/html_css.asp) query

Before cleaning, the code retrieved is of the following form, which matches with what is on the website.
```
<h1 style="color:blue;">A Blue Heading</h1> <p style="color:red;">A red paragraph.</p>
```
And since it should be escaped, the expected escaped form is
```
&lt;h1 style=&quot;color:blue;&quot;&gt;A Blue Heading&lt;/h1&gt; &lt;p style=&quot;color:red;&quot;&gt;A red paragraph.&lt;/p&gt;
```
However, after Jsoup.clean, the non escaped code is in fact
```
A Blue Heading 
<p>A red paragraph.</p>
```
Which, when escaping is then applied to it, turns into
```
A Blue Heading 
&lt;p&gt;A red paragraph.&lt;/p&gt;
```
["css flexbox"](https://www.w3schools.com/css/css3_flexbox.asp) query is another test. It should be producing 
```
<div class="flex-container"> <div>1</div> <div>2</div> <div>3</div> </div>
```
But after clean, instead, it is just `1 2 3`

Seems like HTML escaping is doing its job, but Jsoup.clean is stripping away tags that it shouldn't be.

Reply:
https://jsoup.org/apidocs/org/jsoup/safety/Whitelist.html#basicWithImages()

Yeah, clean removes all tags except a whitelist, and that whitelist does not contain div.

Reply:
Ah, my mistake. I'll change the whitelist param.